BraintreeCore/src/main/java/com/braintreepayments/api/core/AnalyticsClient.kt

```diff
@@ -24,8 +24,13 @@ class AnalyticsClient internal constructor(
     private val analyticsParamRepository: AnalyticsParamRepository = AnalyticsParamRepository.instance,
     private val time: Time = Time(),
     private val configurationLoader: ConfigurationLoader = ConfigurationLoader.instance,
-    private val merchantRepository: MerchantRepository = MerchantRepository.instance
+    private val merchantRepository: MerchantRepository
 ) {
+
+    constructor() : this(
+        merchantRepository = MerchantRepository.instance
+    )
```

Comment:
The AnalyticsClient constructor needs to be defined as public (still scoped to the library) for the Launcher classes to instantiate them while keeping the default constructor internal to not leak internal classes. The only way I found to create an empty constructor is to pass one of the public params to the default constructor. Would love to hear if anyone has other ideas for solving this issue.